### PR TITLE
Define yaml configuration interface.

### PR DESCRIPTION
**Pull request by annymsMthd**
(+semver: feature)

Signed-off-by: Joshua Benjamin <joshuabnjmn@gmail.com>

### CODE REVIEW

**Comment by annymsMthd** [comment]
Just tossing out my ideas for the configuration for the generator.

File: docs/examples/components/details.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+componentName: details
+description: The details component handles the flow for device details.
+
+emitters:
+  - message: kafmesh.deviceId.detail
```
Comment (cuevaskoch) [inline docs/examples/components/details.yaml]: this is both protobuf schema info and kafka topic right?

Comment (annymsMthd) [reply docs/examples/components/details.yaml]: yeah. I'm thinking of making a topic name override in case you wan to break out of the mold

**Comment by annymsMthd** [comment]
Okay WIP removed. I think I've finalized the initial yaml interface

**Pull request merged**